src/core/ext/filters/load_reporting/server_load_reporting_filter.cc

```diff
@@ -287,12 +158,99 @@ const char* ServerLoadReportingCallData::GetStatusTagForStatus(
       return grpc::load_reporter::kCallStatusClientError;
   }
 }
+}  // namespace
+
+grpc_core::ArenaPromise<grpc_core::ServerMetadataHandle>
+ServerLoadReportingFilter::MakeCallPromise(
+    grpc_core::CallArgs call_args,
+    grpc_core::NextPromiseFactory next_promise_factory) {
+  grpc_core::Slice service_method;
+  if (const grpc_core::Slice* path =
+          call_args.client_initial_metadata->get_pointer(
+              grpc_core::HttpPathMetadata())) {
+    service_method = path->Ref();
+  }
+  std::string target_host;
+  if (const grpc_core::Slice* authority =
+          call_args.client_initial_metadata->get_pointer(
+              grpc_core::HttpAuthorityMetadata())) {
+    target_host = absl::AsciiStrToLower(authority->as_string_view());
+  }
+  std::string client_ip_and_lr_token;
+  auto lb_token =
+      call_args.client_initial_metadata->Take(grpc_core::LbTokenMetadata())
+          .value_or(grpc_core::Slice());
+  client_ip_and_lr_token = MakeClientIpAndLrToken(
+      lb_token.as_string_view(), call_args.client_initial_metadata);
+  opencensus::stats::Record(
+      {{::grpc::load_reporter::MeasureStartCount(), 1}},
+      {{::grpc::load_reporter::TagKeyToken(),
+        {client_ip_and_lr_token.data(), client_ip_and_lr_token.length()}},
+       {::grpc::load_reporter::TagKeyHost(),
+        {target_host.data(), target_host.length()}},
+       {::grpc::load_reporter::TagKeyUserId(),
+        {peer_identity_.data(), peer_identity_.length()}}});
+  return grpc_core::ArenaPromise<grpc_core::ServerMetadataHandle>(
```

Comment:
Might be helpful to describe steps in promise

Reply:
Added some commentary